src/applications/vaos/new-appointment/components/CommunityCareProviderSelectionPage/ProviderList.jsx

```diff
@@ -0,0 +1,275 @@
+import React from 'react';
+import { useDispatch } from 'react-redux';
+import LoadingIndicator from '@department-of-veterans-affairs/component-library/LoadingIndicator';
+import recordEvent from 'platform/monitoring/record-event';
+import {
+  FETCH_STATUS,
+  FACILITY_SORT_METHODS,
+  GA_PREFIX,
+} from '../../../utils/constants';
+import { updateCCProviderSortMethod } from '../../redux/actions';
+import ResidentialAddress from '../../../components/ResidentialAddress';
+import InfoAlert from '../../../components/InfoAlert';
+import LoadProvidersErrorAlert from './LoadProvidersErrorAlert';
+import NoProvidersAlert from './NoProvidersAlert';
+
+export default function ProviderList({
+  address,
+  checkedProvider,
+  communityCareProviderList,
+  idSchema,
+  initialProviderDisplayCount,
+  loadingLocations,
+  loadingProviders,
+  onChange,
+  providersListLength,
+  requestLocationStatus,
+  requestStatus,
+  setCheckedProvider,
+  setProvidersListLength,
+  setShowProvidersList,
+  sortMethod,
+  typeOfCareName,
+}) {
+  const dispatch = useDispatch();
+  const currentlyShownProvidersList = communityCareProviderList?.slice(
+    0,
+    providersListLength,
+  );
+  const sortByDistanceFromResidential =
+    !sortMethod || sortMethod === FACILITY_SORT_METHODS.distanceFromResidential;
+
+  const sortByDistanceFromCurrentLocation =
+    sortMethod === FACILITY_SORT_METHODS.distanceFromCurrentLocation;
+  return (
+    <div className="vads-u-background-color--gray-lightest vads-u-padding--2 medium-screen:vads-u-padding--3">
+      <h2
+        id="providerSelectionHeader"
+        className="vads-u-font-size--h3 vads-u-margin-top--0"
+      >
+        Choose a provider
+      </h2>
+      {!loadingLocations &&
+        sortByDistanceFromResidential && (
+          <>
+            {requestLocationStatus !== FETCH_STATUS.failed && (
+              <>
+                <p className="vads-u-margin-top--0 vads-u-margin-bottom--2">
```

Comment:
Thought about breaking these sorted by messages into a separate component but they seem distinct enough to leave as is.